src/rust/sdk/src/lib.rs

```diff
@@ -0,0 +1,253 @@
+use anyhow::{anyhow, bail, Context, Result};
+use aptos_api_types::{MoveModuleId, TransactionInfo, UserTransactionRequest, U64};
+use aptos_sdk::crypto::ed25519::Ed25519PrivateKey;
+use aptos_sdk::crypto::ValidCryptoMaterialStringExt;
+use aptos_sdk::move_types::language_storage::TypeTag;
+use aptos_sdk::rest_client::aptos::Balance;
+use aptos_sdk::rest_client::{Client, Resource};
+use aptos_sdk::types::account_address::AccountAddress;
+use aptos_sdk::types::chain_id::ChainId;
+use aptos_sdk::types::{AccountKey, LocalAccount};
+use econia_types::events::EconiaEvent;
+use reqwest::Url;
+use serde::Deserialize;
+use std::collections::HashMap;
+use std::fmt::Debug;
+use std::fs::File;
+use std::str::FromStr;
+
+use tx_builder::EconiaTransactionBuilder;
+
+mod tx_builder;
+
+pub const SUBMIT_ATTEMPTS: u8 = 10;
+
+#[derive(Deserialize, Debug, Clone)]
+struct AptosConfig {
+    private_key: String,
+    account: String,
+}
+
+type AptosConfigYaml = HashMap<String, HashMap<String, AptosConfig>>;
+
+impl AptosConfig {
+    pub fn from_config(path: &str, profile_name: &str) -> Self {
+        let file = File::open(path).expect("invalid config path provided");
+        let config =
+            serde_yaml::from_reader::<File, AptosConfigYaml>(file).expect("config file is invalid");
+        let profiles = config
+            .get("profiles")
+            .expect("profiles section missing in config file");
+        profiles
+            .get(profile_name)
+            .expect("given profile name is missing in config file")
+            .clone()
+    }
+}
+
+pub struct EconiaTransaction {
+    pub info: TransactionInfo,
+    pub request: UserTransactionRequest,
+    pub events: Vec<EconiaEvent>,
+    pub timestamp: U64,
+}
+
+pub struct EconiaClient {
+    econia: AccountAddress,
+    econia_module: MoveModuleId,
+    aptos: Client,
```

Comment:
Fields should be named so that it is clear what their contents are. It's impossible to guess what a field would contain if they have names like `econia` and `aptos`.